apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/OperationExtensions.kt

```diff
@@ -134,11 +137,43 @@ fun <D : Operation.Data> Operation<D>.parse(
 }
 
 /**
- * Parses GraphQL operation raw response from the [byteString] with provided [customScalarAdapters] and returns result [Response]
+ * Parses GraphQL operation raw response from [string] with provided [customScalarAdapters] and returns result [Response]
  */
 fun <D : Operation.Data> Operation<D>.parse(
     string: String,
     customScalarAdapters: CustomScalarAdapters = DEFAULT
 ): Response<D> {
   return parse(Buffer().writeUtf8(string), customScalarAdapters)
 }
+
+/**
+ * Parses GraphQL operation raw response from [map] with provided [customScalarAdapters] and returns result [Response]
+ *
+ * @param map: a [Map] representing the response. It typically include a "data" field
+ */
+fun <D : Operation.Data> Operation<D>.parse(
+    map: Map<String, Any?>,
```

Comment:
just a thought if introducing type alias for this type argument in `MapResponseReader` will be more readable

Reply:
We could but that would use another name and there's a finite about of them 😅 . We use `RecordSet` in some places but TBH I quite dislike it (as I always think of it as a Set<Record>, which it is not!). 

Since this map should be the result of `JsonReader.readRecursively()`, I was considering adding `JsonXYZ` aliases at some point. `Map<String, Any?>` would then be a JsonObject, or maybe a `ApolloJsonObject`, or just `AJsonObject`. But I'm not sure  how that would unfold yet.